app/js/app/services/Auth.js

```diff
@@ -96,12 +101,18 @@ define([], function() {
 				}
 				
 				userResource.$promise.then(function(u) {
-					$timeout(function() {
-						$rootScope.user = u;
-						setupUserConsistencyCheck();
-						$rootScope.$apply();
+
+					return new Promise(function(resolve, reject) {
+						$timeout(function() {
+							$rootScope.user = u;
+							setupUserConsistencyCheck();
+							$rootScope.$apply();
+							resolve();
+						});
+					}).then(updateUserPreferences).then(function() {
```

Comment:
This chained `.then(...)` business is awful and we hate it, but Ian and I didn't want to rewrite the whole login flow just yet!